fe/fe-core/src/main/java/org/apache/doris/planner/HashDistributionPruner.java

```diff
@@ -119,9 +130,6 @@ public Collection<Long> prune(int columnId, PartitionKey hashKey, int complex) {
             Collection<Long> subList = prune(columnId + 1, hashKey, newComplex);
             resultSet.addAll(subList);
             hashKey.popColumn();
-            if (resultSet.size() >= bucketsList.size()) {
```

Comment:
why remove this

Reply:
In the case of `IN` query, we need to collect all the pruned buckets for the data in the `IN list`. When `resultSet.size() >= bucketsList.size()` is `true`, it is possible that only one data in the `IN list` has been processed, that is, the number of buckets is less than the length of `IN list`. So maybe we can keep the `if`, and change it to this:
```c++
if ( isPointQuery() && resultSet.size() >= bucketsList.size()) {
  break;
}
```